File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
   bool isPalindrome(int x) {
        string st=to_string(x);
        
        int n=st.size();
        string left="";
        string right="";

        for(int i=0;i<n;i++) left+=st[i];

        for(int i=n-1;i>=0;i--) right+=st[i];

        if(left==right) return true;
        
        else return false;
        
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int num = x;
        int rev = 0;
        while (num != 0 && num > 0) {
    rev = rev * 10 + num % 10; // Add last digit
    num = num / 10;            // Remove last digit
}
        if(rev == x){
            return true;
        }else{
            return false;
        }
    }
    

}